src/transformStateWithClones.js

```diff
@@ -8,6 +8,32 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateArray = [];
+  const stateModif = { ...state };
+
+  for (const i of actions) {
+    if (i.type === 'addProperties') {
+      Object.assign(stateModif, i.extraData);
+      stateArray.push({ ...stateModif });
+      continue;
+    }
+
+    if (i.type === 'removeProperties') {
+      for (const key of i.keysToRemove) {
+        delete stateModif[key];
+      }
+      stateArray.push({ ...stateModif });
+      continue;
+    }
+
+    if (i.type === 'clear') {
+      Object.keys(stateModif).forEach((key) => delete stateModif[key]);
+      stateArray.push({ ...stateModif });
+      continue;
+    }
```

Comment:
[CODE STYLE]: Use a switch statement instead of multiple if statements when you have a limited amount of conditions to make your code cleaner and more maintainable.